### ISSUE TITLE
string.printable.isprintable() returns False

### ISSUE THREAD

**Issue by 228b5382-0e4b-45c6-90d4-2db145488412** [python/cpython#67206]
BPO | [23017](https://bugs.python.org/issue23017)
--- | :---
Nosy | @birkenfeld, @vstinner, @ezio-melotti, @stevendaprano, @bitdancer, @4kir4, @iritkatriel
Files | <li>[bug-string-ascii.py](https://bugs.python.org/file37391/bug-string-ascii.py "Uploaded as text/plain at 2014-12-09.03:51:59 by planet36"): Test case shows that string.printable has control characters</li><li>[0001-Fix-string.printable-respect-POSIX-spec.patch](https://bugs.python.org/file37398/0001-Fix-string.printable-respect-POSIX-spec.patch "Uploaded as text/plain at 2014-12-09.14:42:29 by bru")</li><li>[docs-string.printable.diff](https://bugs.python.org/file37441/docs-string.printable.diff "Uploaded as text/plain at 2014-12-13.15:30:05 by @4kir4")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2014-12-09.03:52:01.009>
labels = ['type-bug', '3.9', '3.10', '3.11', 'library', 'expert-unicode', 'docs']
title = 'string.printable.isprintable() returns False'
updated_at = <Date 2021-11-29.16:17:13.755>
user = 'https://bugs.python.org/planet36'
```

bugs.python.org fields:
```python
activity = <Date 2021-11-29.16:17:13.755>
actor = 'iritkatriel'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Documentation', 'Library (Lib)', 'Unicode']
creation = <Date 2014-12-09.03:52:01.009>
creator = 'planet36'
dependencies = []
files = ['37391', '37398', '37441']
hgrepos = []
issue_num = 23017
keywords = ['patch']
message_count = 5.0
messages = ['232343', '232376', '232382', '232613', '407290']
nosy_count = 10.0
nosy_names = ['georg.brandl', 'vstinner', 'ezio.melotti', 'steven.daprano', 'r.david.murray', 'docs@python', 'akira', 'planet36', 'bru', 'iritkatriel']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue23017'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

### Linked PRs
* gh-128820
* gh-128867
* gh-128868

**Comment by picnixz** [comment]
Still reproducible on main.

**Comment by vstinner** [comment]
> We could document that this does not match the posix definition of printable. It also does not match the RFC 5822 definition of printable (for example), which does *not* include whitespace characters (not even space), but the posix definition is a more likely source of confusion.
> (...)
> For backward compatibility reasons I suspect we are stuck with the discrepancy, but perhaps others will think it worth the pain of changing string.printable. I kind of doubt it, though.

I agree that it would be a bad idea to change `string.printable` value now. It's too late. The best that we can do is to document that `string.printable.isprintable()` is false but it's deliberate.

**Comment by picnixz** [comment]
We documented and backported the behaviour, hence I will close this issue as `completed`.
